src/main/java/net/openhft/chronicle/bytes/AbstractBytes.java

```diff
@@ -53,12 +55,12 @@
     private boolean lenient = false;
     private boolean lastNumberHadDigits = false;
 
-    AbstractBytes(@NotNull BytesStore<Bytes<Underlying>, Underlying> bytesStore, long writePosition, long writeLimit)
+    AbstractBytes(@NotNull(exception = NullPointerException.class) BytesStore<Bytes<Underlying>, Underlying> bytesStore, long writePosition, long writeLimit)
```

Comment:
I understand why all of these were done, but it does not feel ideal - it adds a lot of extra code, and reduces the signal/noise ration. 

It just feels strange that both Intellij and the [maven plugin we use](https://github.com/osundblad/intellij-annotations-instrumenter-maven-plugin/) throw `IllegalArgumentException` instead of NPE. As you have pointed out [in this issue](https://github.com/osundblad/intellij-annotations-instrumenter-maven-plugin/issues/53) it would seem more useful that this instrumentation should mimic `Objects::requireNonNull` (or our `ObjectUtils::requireNotNull`) and throw a NPE. 

Another approach we could follow is:
* implement [issue 53](https://github.com/osundblad/intellij-annotations-instrumenter-maven-plugin/issues/53) ourselves so we can configure the plugin to throw NPE (in one place - our root-parent-pom)
* (as it does not appear to be possible to configure Intellij's non-null exception globally) so we could turn off the IDE's instrumentation and commit that config to git (for all projects)

Reply:
In the short term: could we change the tests so that they are happy to accept either NPE or IAE? Then we could get phase 1 of this PR in without the changing all the `@NonNull` annotations?

Reply:
I agree the annotation param adds a lot of noise, it would be great to find a way to avoid this.

Reply:
Great feedback. The simplest and best solution might be to turn off this feature in IDEA. This means that only real assertions are being made regardless if run under IDEA or Maven and we get consistent behavior and reduced SNR.

We could add a specific test that will trigger unless the feature is turned off and that will output a resolution.

![image](https://user-images.githubusercontent.com/7457876/139640910-1769651c-9332-48af-a4eb-4b47343c89c2.png)